src/Sarif/Core/Result.cs

```diff
@@ -120,8 +120,15 @@ public bool TryIsSuppressed(out bool isSuppressed)
 
             // If the status of any of the suppressions is "underReview" or "rejected",
             // then the result should not be considered suppressed. Otherwise, the result should be considered suppressed.
-            // https://github.com/microsoft/sarif-tutorials/blob/main/docs/Displaying-results-in-a-viewer.md#determining-suppression-status
+            // https://github.com/microsoft/sarif-tutorials/blob/main/docs/Displaying-results-in-a-viewer.md#determining-suppression-status           
             isSuppressed = !suppressions.Any(s => s.Status == SuppressionStatus.UnderReview || s.Status == SuppressionStatus.Rejected);
+
+            // if it is suppressed and are expired            
+            if (isSuppressed && checkExpired)
+            {
+                isSuppressed = suppressions.Any(s => (s.TryGetProperty("expiryUtc", out DateTime expiryUtc) && expiryUtc > DateTime.UtcNow) && s.Status == SuppressionStatus.Accepted);
```

Comment:
>expiryUtc [](http://example.com/codeflow?start=72&length=9)

I think SARIF spec is going to add some properties including "ExpiryUtc" or similar to Suppression entity? cc @eddynaka @michaelcfanning @EasyRhinoMSFT

Reply:
sounds good to me